src/main/java/org/ambraproject/wombat/service/ParseReferenceService.java

```diff
@@ -120,11 +120,25 @@ public String getValue() {
       NodeList extLinkList = element.getElementsByTagName("ext-link");
       if (!ParseXmlUtil.isNullOrEmpty(extLinkList)) {
         Element extLink = (Element) extLinkList.item(0);
+        // use the ext-link as doi, only if ext-link previous text is "doi:" or "doi"
+        boolean useDoi = false;
+        Node previousNode = extLink.getPreviousSibling();
+        if (previousNode != null) {
+          String previousText = extLink.getPreviousSibling().getTextContent();
```

Comment:
I suggest using `previousNode`, since it's already assigned as `extLink.getPreviousSibling()`.